packages/js/src/notifications/helpers.ts

```diff
@@ -173,6 +182,88 @@ export const unarchive = async ({
   }
 };
 
+export const snooze = async ({
+  emitter,
+  apiService,
+  args,
+}: {
+  emitter: NovuEventEmitter;
+  apiService: InboxService;
+  args: SnoozeArgs;
+}): Result<Notification> => {
+  const { notificationId, optimisticValue } = getNotificationDetails(
+    args,
+    {
+      isSnoozed: true,
+      snoozedUntil: args.snoozeUntil,
+      isRead: false,
+      readAt: null,
+    },
+    {
+      emitter,
+      apiService,
+    }
+  );
+
+  try {
+    emitter.emit('notification.snooze.pending', {
+      args,
+      data: optimisticValue,
+    });
+
+    const response = await apiService.snooze(notificationId, args.snoozeUntil);
+
+    const updatedNotification = new Notification(response, emitter, apiService);
+    emitter.emit('notification.snooze.resolved', { args, data: updatedNotification });
+
+    return { data: updatedNotification };
+  } catch (error) {
+    emitter.emit('notification.snooze.resolved', { args, error });
+
+    return { error: new NovuError('Failed to snooze notification', error) };
+  }
+};
+
+export const unsnooze = async ({
+  emitter,
+  apiService,
+  args,
+}: {
+  emitter: NovuEventEmitter;
+  apiService: InboxService;
+  args: UnsnoozeArgs;
+}): Result<Notification> => {
+  const { notificationId, optimisticValue } = getNotificationDetails(
+    args,
+    {
+      isSnoozed: false,
+      snoozedUntil: null,
```

Comment:
according to the API changes we have to update these as well:
```
createdAt: now,
read: false,
lastReadDate: null,
```

Reply:
I updated the API, basically manual unsnooze or cancel of snooze, should revert to the original notif state before snooze, so basically it changes just this on both snooze/unsnooze on API lvl:
```
  updatePayload = {
    snoozedUntil,
    seen: true,
    lastSeenDate: new Date(),
    archived: false,
    archivedAt: null,
  };
```